src/main/java/org/scijava/convert/DefaultConverter.java

```diff
@@ -168,10 +169,20 @@ public <T> T convert(final Object src, final Class<T> dest) {
 		}
 		if (saneDest == String.class) {
 			// destination type is String; use Object.toString() method
-			final String sValue = src.toString();
-			@SuppressWarnings("unchecked")
-			final T result = (T) sValue;
-			return result;
+			if (src.getClass().isArray()) {
+				final String elementString = ArrayUtils.toCollection(src).stream() //
+					.map(object -> convert(object, String.class)) //
+					.collect(Collectors.joining(","));
+				@SuppressWarnings("unchecked")
+				final T result = (T) elementString;
```

Comment:
I think the array should be prefixed by `{` and suffixed by `}`. Otherwise:
```java
String[][] s = { {"hello", "there"}, {"goodbye", "now"} };
```
becomes:
```
hello, there, goodbye, now
```
And it's not obvious that it was once nested arrays.

And even then, other cases will be hosed upon reconstitution, such as strings containing `,` or `{` or `}` characters. We would need to perform escaping/unescaping to avoid that.

Based on the unit test below, I assume the primary use case here is array persistence, so then the escaping is necessary, so that strings converted from arrays are convertible back to an equivalent array object afterward. Right? In that case, it would be better to support direct persistence of properly typed arrays, rather than feeding through the converter framework here, don't you think?

Reply:
Great, thanks. I wrote a ton of unit tests that also test nested arrays.

> Based on the unit test below, I assume the primary use case here is array persistence, so then the escaping is necessary, so that strings converted from arrays are convertible back to an equivalent array object afterward. Right? In that case, it would be better to support direct persistence of properly typed arrays, rather than feeding through the converter framework here, don't you think?

Maybe, but would that not be more work?